File: java/src/test/java/com/hachicore/demo/jooq/JooqJoinShortCutTest.java
package com.hachicore.demo.jooq;

import com.hachicore.demo.jooq.film.FilmRepository;
import com.hachicore.demo.jooq.film.FilmWithActors;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
public class JooqJoinShortCutTest {

    @Autowired
    FilmRepository filmRepository;

    @Test
    @DisplayName("implicitPathJoin_테스트")
    void implicitPathJoin_테스트() {

        List<FilmWithActors> original = filmRepository.findFilmWithActorsList(1L, 10L);
        List<FilmWithActors> implicit = filmRepository.findFilmWithActorsListImplicitPathJoin(1L, 10L);

        assertThat(original)
                .usingRecursiveFieldByFieldElementComparator()
                .isEqualTo(implicit);
    }

    @Test
    @DisplayName("explicitPathJoin_테스트")
    void explicitPathJoin_테스트() {

        List<FilmWithActors> original = filmRepository.findFilmWithActorsList(1L, 10L);
        List<FilmWithActors> implicit = filmRepository.findFilmWithActorsListExplicitPathJoin(1L, 10L);

        assertThat(original)
                .usingRecursiveFieldByFieldElementComparator()
                .isEqualTo(implicit);
    }
}

File: kotlin/src/test/kotlin/com/hachicore/demo/jooq/JooqJoinShortCutTest.kt
package com.hachicore.demo.jooq

import com.hachicore.demo.jooq.film.FilmRepository
import com.hachicore.demo.jooq.film.FilmWithActor
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class JooqJoinShortCutTest(
    val filmRepository: FilmRepository
) {

    @Test
    @DisplayName("implicitPathJoin_테스트")
    fun implicitPathJoin_테스트() {
        val original: List<FilmWithActor> = filmRepository.findFilmWithActorList(1L, 10L)
        val implicit: List<FilmWithActor> = filmRepository.findFilmWithActorListImplicitPathJoin(1L, 10L)

        assertThat(original)
            .usingRecursiveFieldByFieldElementComparator()
            .isEqualTo(implicit)
    }

    @Test
    @DisplayName("explicitPathJoin_테스트")
    fun explicitPathJoin_테스트() {
        val original: List<FilmWithActor> = filmRepository.findFilmWithActorList(1L, 10L)
        val implicit: List<FilmWithActor> = filmRepository.findFilmWithActorListExplicitPathJoin(1L, 10L)

        assertThat(original)
            .usingRecursiveFieldByFieldElementComparator()
            .isEqualTo(implicit)
    }
}